### DISCUSSION TITLE
Vercel Development fails after implementation of dynamic routes

### DISCUSSION THREAD

**Discussion by MiejkeHoltermann** [neuefische/questions#718] [Web] [answered]
Hello, my Vercel development failed after I implemented dynamic routes in my app. ChatGPT and internet forums suggest that it has something to do with the routes being rendered client side and Next.js working server side. But we never learned anything like this and we created dynamic routes before. So normally it shouldn't be a problem?
I imported two components (Map.js and Carousel.js) dynamically, so maybe that causes a problem. On the other hand I have the same problem with other dynamic pages that don't use those two components. If I follow the links I can get to any page. But if I reload a page, I get an error like this:

TypeError: Cannot read properties of undefined (reading 'slug')
This error happened while generating the page. Any console logs will be displayed in the terminal window.
Source

pages\[slug]\map.js (20:40) @ slug

  18 | <Header trip={currentTrip} />
  19 | <StyledHeading1>Itinerary</StyledHeading1>
> 20 | <ToggleLink href={`/${currentTrip.slug}/itinerary`}>
     |                                  ^
  21 |   <ToggleLinkImage
  22 |     src="/list.svg"
  23 |     height={40}

**Comment by alperkuertuel** [comment]
I had something similar with my slugs after hard refresh. 
You have to wrap the function or whatever around an `if(!slug){ "What needs to be done! }`
Look at this post from earlier. This solved my problem then ->  https://github.com/orgs/neuefische/discussions/464

Much success! I hope this will work for you...

**Comment by MiejkeHoltermann** [comment, marked as answer]
Ah, thank you, I didn't see your answer, but I already found it myself. Oopsie 😊
